sklearn/utils/_array_api.py

```diff
@@ -103,6 +105,14 @@ def _check_array_api_dispatch(array_api_dispatch):
                 f"NumPy must be {min_numpy_version} or newer to dispatch array using"
                 " the API specification"
             )
+        if os.environ.get("SCIPY_ARRAY_API") != "1":
+            warnings.warn(
+                (
+                    "Some scikit-learn array API features might rely on enabling "
+                    "SciPy's own support for array API to function properly. "
+                ),
+                UserWarning,
+            )
```

Comment:
I took this very literally and I am also not sure if UserWarning is the correct thing to use here, so any advice would be greatly appreciated!